### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by wuyepabdul**
In this milestone, I did the following:
- Created Classroom class with the following:
  - `@label` instance variable,  initialized in the constructor.
  - Setter and getter for `@label` ( using `attr_accessor`).
- Created the `has-many`/`belongs-to` relationship between `Classroom` and `Student`, as well as:
  - Created the `has-many` side (a classroom has many students).
  - Created the `belongs-to` side (a student belongs to a classroom).
  - Made sure that when adding a student to a classroom it also sets the classroom for the student.
  - Made sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Created a class `Book` with the following:
  - `@title` and `@author` instance variables, initialized in the constructor.
  - Setters and getters for instance variables (using `attr_accessor`).
- Created a class `Rental` with the following:
  - `@date` instance variable, initialized in the constructor.
  - Setter and getter for `@date` (using `attr_accessor`).
- Created the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following were be implemented:
  - Created the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Created the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Created the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Created the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modified the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (Ghiftee) [inline book.rb]: - [x] Please use a getter and setter method for rentals also. In other words, kindly add it to `` attr_accessor``

File: rentals.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (Ghiftee) [inline rentals.rb]: - [x] Please use a getter and setter method for book and person also. In other words, kindly add them to `` attr_accessor``

File: person.rb [hunk 3]
```diff
@@ -2,14 +2,15 @@
 
 class Person
   attr_accessor :name, :age, :parent_permission
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (Ghiftee) [inline person.rb]: - [x] Please use a getter and setter method for rentals also. In other words, kindly add it to `` attr_accessor``

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello 👋🏽  @wuyepabdul 

Good job working on the project so far! 
✔️ 
However, there are some issues that you need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._
- [x] In Book class, please create a method for adding rentals. The book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. You can refer to [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for more information on how to achieve this.
- [x] In Person class, please create a method for adding rentals. The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association. You can refer to [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for more information on how to achieve this.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by mikethreels — APPROVED** [review]
Hi Duls, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 :
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by wuyepabdul** [comment]
Thanks for the review @mikethreels  and @Ghiftee. Your suggestions really helped me to understand the concept of associations much better

**Pull request merged**